content/blog/2016/2016-08-29-jenkins-world-speaker-blog-goodgame.adoc

```diff
@@ -0,0 +1,42 @@
+
+The results of this analysis can be used by all people involved working with Jenkins. To achieve this, I developed a simple extension of SonarQube, containing everything which is needed to hook up our SonarQube with our Jenkins-environment. The implementation contains a new basic-language “Jenkins“ and an initial set of rules were defined.
+
+Of course the needs depend strongly on the way Jenkins is being used, so not every rule implemented might be useful for every team, but this applies as well all other code-analysis. The main inspiration for the rules were developer feedback and some articles found in the web. The different possibilities to use and configure Jenkins provides a lot of potential for many more rules. With this new approach of quality-analysis, we can enforce best practices like:
+
+* Polling must die (Rather trigger a build due to pushes instead of poll the repository every x minutes)
+* Use Log Rotator (Not using log-rotator can result in disk space problems on the master)
+* Use slaves/labels (Jobs should be defined where to run)
+* Don’t build on the master (In larger systems, don't build on the master)
+* Enforce plugin usage (For example: Timestamp, Mask-Passwords)
+* Naming sanity (Limit project names to a sane (e.g. alphanumeric) character set)
+* Analyze Groovy Scripts (For example: Prevent System.exit(0) in System Groovy Scripts)
+
+image:/images/post-images/jw-speaker-blog-goodgame/jenkins1.png[role=left]
+
+Besides taking control over all configuration of any Jenkins-instance we want, there is also room for additional metrics, like measuring the amount and different types of jobs (Freestyle/Maven etc...) to get an overview about the general load of the Jenkins-instance. A more sophisticated idea is to measure complexity of jobs and even pipelines. As code, jobs-configuration gets harder to understand the more steps are involved. On the one hand scripts, conditions and many parameters can negatively influence the readability, especially if you have external dependencies (like scripts) in different locations. On the other hand, pipelines can also grow very complex when many jobs are involved and chained for execution. It will be very interesting for us too see where and why too complex pipelines are being created.
```

Comment:
"taking control of"
"Jenkins instance"
"jobs configuration"
s/too see/to see/